.github/workflows/swift_matrix.yml

```diff
@@ -108,6 +112,16 @@ jobs:
       if: ${{ matrix.swift.enabled }}
     # https://github.com/actions/checkout/issues/766
       run: git config --global --add safe.directory ${GITHUB_WORKSPACE}
+
+    - name: Restore .build
+      if: ${{ matrix.swift.enabled && (inputs.cache_key_prefix != '') && !(github.run_attempt > 1) }}
```

Comment:
`!(github.run_attempt > 1)` will be useful when something is inconsistent in CI and you're retrying the job.
Although you can delete cache from the GitHub Actions UI too, on web.